contribs/gnostats/agent/agent.go

```diff
@@ -0,0 +1,81 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/gnolang/gnostats/proto"
+)
+
+// config holds both the injected Gno node RPC client and the Hub gRPC client.
+type config struct {
+	hClient      proto.HubClient
+	rClient      rpcClient
+	pollInterval time.Duration // Minimum time interval between two data points
+}
+
+type agent struct {
+	cfg    config
+	cancel context.CancelFunc
+}
+
+// Start registers with the Hub using Gno node static info, then pushes dynamic
+// info from the Gno node to the Hub at intervals specified by pollInterval
+func (a *agent) Start(ctx context.Context) error {
+	// Store a cancelFunc to make the agent stoppable using the Stop() method
```

Comment:
I don't think you need to create a wrapped context, since you control the agent through the context you already give it. As the caller I'm expecting that when I cancel the context I kill the agent